### PR TITLE
gh2gcs: add download only option

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

gh2gcs: add download only option

#### Which issue(s) this PR fixes:

Fixes Partially  https://github.com/kubernetes/release/issues/1320

#### Does this PR introduce a user-facing change?

```release-note
gh2gcs: add download only option
```

### CODE REVIEW

File: cmd/gh2gcs/cmd/root.go [hunk 1]
```diff
@@ -136,6 +138,13 @@ func init() {
 		"local directory for releases to be downloaded to",
 	)
 
+	rootCmd.PersistentFlags().BoolVar(
+		&opts.downloadOnly,
+		downloadOnlyFlag,
+		false,
+		"download only the releases, with this flag the files will not be pushed to GCS and you need to set the output-dir flag as well",
```
Comment (hasheddan) [inline cmd/gh2gcs/cmd/root.go:145]: nitpick: how do you feel about this rewording? The current wording felt a little confusing to me, but that could just be me so feel free to ignore suggestion if you don’t see it as an improvement :)

```suggestion
		"only download the releases, do not push them to GCS. Requires the output-dir flag to also be set",
```

Comment (cpanato) [reply cmd/gh2gcs/cmd/root.go:145]: thanks for your review Daniel!
done

File: cmd/gh2gcs/cmd/root.go [hunk 2]
```diff
@@ -198,11 +207,17 @@ func run(opts *options) error {
...
-			return errors.Wrap(err, "uploading release assets to GCS")
+		if !opts.downloadOnly {
+			if err := gh2gcs.Upload(&rc, gh, opts.outputDir); err != nil {
+				return errors.Wrap(err, "uploading release assets to GCS")
+			}
 		}
 	}
 
+	if opts.downloadOnly {
+		logrus.Infof("Files downloaded to %s folder", opts.outputDir)
```
Comment (saschagrunert) [inline cmd/gh2gcs/cmd/root.go:218]: We probably can log this in any case. I would just move it directly after the call to `DownloadReleases`

Comment (justaugustus) [reply cmd/gh2gcs/cmd/root.go:218]: +1

Comment (cpanato) [reply cmd/gh2gcs/cmd/root.go]: done! thanks for the review!

File: cmd/gh2gcs/cmd/root.go [hunk 3]
```diff
@@ -197,9 +206,12 @@ func run(opts *options) error {
 		if err := gh2gcs.DownloadReleases(&rc, gh, opts.outputDir); err != nil {
 			return errors.Wrap(err, "downloading release assets")
 		}
+		logrus.Infof("Files downloaded to %s folder", opts.outputDir)
```
Comment (saschagrunert) [inline cmd/gh2gcs/cmd/root.go:209]: One of the nittiest nits and non blocking: `folder` seems to be a term from the windows world. I would prefer if we stick to `directory` instead. 😋

**Comment by justaugustus** [comment]
/lgtm
/approve

Feel free to fix @saschagrunert's nit in a follow-up.

File: cmd/gh2gcs/cmd/root.go (hunk 3 above)
Comment (cpanato) [reply cmd/gh2gcs/cmd/root.go:209]: this shows my age lol from the 3.11 for workgroups

**Pull request merged**